xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1655,7 +1655,7 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
         st->bitDepth = 8;
         const AVPixFmtDescriptor* desc =
             av_pix_fmt_desc_get(static_cast<AVPixelFormat>(pStream->codecpar->format));
-        if (desc != nullptr && desc->comp != nullptr)
+        if (desc != nullptr)
```

Comment:
i was wondering how the compiler would know that `desc->comp` cannot be nullptr. likely because the underlying struct `AVComponentDescriptor` is treated like an array. can we change `desc->comp[0].depth` below to `desc->comp->depth` or would this be just more confusing?

Reply:
`AVComponentDescriptor comp[4];` isn't a pointer and so can never equal `nullptr`.